src/Analyzers/Core/Analyzers/NamingStyle/NamingStyleDiagnosticAnalyzerBase.cs

```diff
@@ -108,7 +108,9 @@ void SyntaxNodeAction(SyntaxNodeAnalysisContext syntaxContext)
             ConcurrentDictionary<Guid, ConcurrentDictionary<string, string?>> idToCachedResult,
             CancellationToken cancellationToken)
         {
-            if (string.IsNullOrEmpty(symbol.Name))
+            if (string.IsNullOrEmpty(symbol.Name) ||
+                // Heuristic for recognizing entry points.
+                symbol.IsStatic && symbol.Name == WellKnownMemberNames.EntryPointMethodName)
```

Comment:
we have an existing helper for this: IsEntryPoint

can you extrac tthis out (an extension on IMethodSymbol would be fine) and use the same for both?

Reply:
yeah, [this method](https://github.com/dotnet/roslyn/blob/main/src/Analyzers/Core/Analyzers/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs#L702-L708) should be extracted out to a general helper and used here.